Doc/library/urllib.parse.rst

```diff
@@ -543,6 +547,53 @@ operating on :class:`bytes` or :class:`bytearray` objects:
 
    .. versionadded:: 3.2
 
+Special URL Behaviors and Scheme Flags
+--------------------------------------
+
+:mod:`urllib.parse` recognizes three special properties of URLs, namely relative
+addressing (used in, for instance, the ``ftp``, ``http``, or ``gopher``
+protocols), netloc-sensitive resolution (used in the ``ftp``, ``http``, or
+``git`` protocols), and URLs that may contain parameters (for instance, ``ftp``
+or ``telnet``).
+
+Relative addressing allows resolution of relative URLs, and netloc-sensitive
+addressing allows resolution with respect to the netloc (domain name) of a URL.
+As HTTP URLs have both behaviors by default, this is demonstrated in the
+following example:
+
+   >>> from urllib.parse import urljoin
+   >>> urljoin('http://example.org/post/x/', '../y/')
+   'http://example.org/post/y/'
+
+Additionally, if it is not indicated that a URL is sensitive to parameters
+(those specified after a semicolon in the path), then they'll be treated as part
+of the path rather than as a distinct component.
+
+Without specifying optional parameters or modifying global variables, Python
+will guess what parameters to apply based on the scheme. Schemes associated with
+each are specified by three lists in :mod:`urllib.parse`:
+
+* ``urllib.uses_relative``
+* ``urllib.uses_netloc``
+* ``urllib.uses_params``
+
+In addition, any function that takes a ``classes`` parameter (for instance,
+:func:`urlparse` and :func:`urljoin`) may override the behavior of the uses
+lists, for instance, parsing a custom or widely unused scheme with the same
+behavior as that of HTTP:
+
+   >>> from urllib.parse import urljoin, SchemeFlag
+   >>> urljoin(
+   ...     'my-protocol://example.org/post/x/', '../y/',
+   ...     flags=(SchemeFlag.NETLOC | SchemeFlag.RELATIVE))
+   'my-protocol://example.org/post/y/'
+
+For reference, the following three scheme classes are present (exactly
+corresponding to the uses lists):
+
+* ``urllib.SchemeFlag.RELATIVE``
+* ``urllib.SchemeFlag.NETLOC``
+* ``urllib.SchemeFlag.PARAMS``
```

Comment:
Add `parse` after `urllib` and Remove `SchemeFlag` portion enums will be available at the global level.

Also, add `urllib.parse.UNIVERSAL` and explain (somewhere) what it does.